Files/Filesystem/Search/SearchSettings.cs

```diff
@@ -0,0 +1,213 @@
+﻿using Microsoft.Toolkit.Mvvm.ComponentModel;
+using Microsoft.Toolkit.Uwp;
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Collections.Specialized;
+using System.ComponentModel;
+using System.Linq;
+
+namespace Files.Filesystem.Search
+{
+    public interface ISearchSettings
+    {
+        ISearchLocation Location { get; }
+        ISearchFilter Filter { get; }
+    }
+
+    public interface ISearchLocation : INotifyPropertyChanged
+    {
+        bool SearchInSubFolders { get; set; }
+    }
+
+    public interface ISearchFilter
+    {
+        string ToAdvancedQuerySyntax();
+    }
+
+    public interface ISearchFilterCollection : IList<ISearchFilter>, ISearchFilter, IHeader, INotifyPropertyChanged, INotifyCollectionChanged
+    {
+    }
+
+    public interface IDateRangeFilter : ISearchFilter, IHeader
+    {
+        DateRange Range { get; }
+    }
+    public interface ISizeRangeFilter : ISearchFilter, IHeader
+    {
+        SizeRange Range { get; }
+    }
+
+    public interface IHeader
+    {
+        string Glyph { get; }
+        string Title { get; }
+        string Description { get; }
+    }
+
+    public class SearchSettings : ObservableObject, ISearchSettings
+    {
+        public static SearchSettings Instance { get; } = new();
```

Comment:
Can you use `IServiceProvider`? (Declare in App.xaml.cs)

Reply:
It is committed.